omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -0,0 +1,1898 @@
+    def __init__(self, robot, robot_copy, robot_copy_type="original"):
+        self.robot = robot
+        self.robot_copy = robot_copy
+        self.robot_copy_type = robot_copy_type if robot_copy_type in robot_copy.prims.keys() else "original"
+        self.disabled_collision_pairs_dict = {}
+
+    def __enter__(self):
+        self._assemble_robot_copy()
+        self._construct_disabled_collision_pairs()
+        return self 
+
+    def __exit__(self, *args):
+        self._set_prim_pose(self.robot_copy.prims[self.robot_copy_type], self.robot_copy.reset_pose[self.robot_copy_type])
+
+    def _assemble_robot_copy(self):
+        if TORSO_FIXED:
+            fk_descriptor = "left_fixed"
+        else:
+            fk_descriptor = "combined" if "combined" in self.robot.robot_arm_descriptor_yamls else self.robot.default_arm
+        self.fk_solver = FKSolver(
+            robot_description_path=self.robot.robot_arm_descriptor_yamls[fk_descriptor],
+            robot_urdf_path=self.robot.urdf_path,
+        )
+        arm_links = self.robot.manipulation_link_names
+
+        if TORSO_FIXED:
+            assert self.arm == "left", "Fixed torso mode only supports left arm!"
+            joint_control_idx = self.robot.arm_cotnrol_idx["left"]
+            joint_pos = np.array(self.robot.get_joint_positions()[joint_control_idx])
+        else:
+            joint_combined_idx = np.concatenate([self.robot.trunk_control_idx, self.robot.arm_control_idx[fk_descriptor]])
+            joint_pos = np.array(self.robot.get_joint_positions()[joint_combined_idx])
+        link_poses = self.fk_solver.get_link_poses(joint_pos, arm_links)
+
+        # Set position of robot copy root prim
+        self._set_prim_pose(self.robot_copy.prims[self.robot_copy_type], self.robot.get_position_orientation())
+
+        # Assemble robot meshes
+        for link_name, meshes in self.robot_copy.meshes[self.robot_copy_type].items():
+            for mesh_name, copy_mesh in meshes.items():
+                # Skip grasping frame (this is necessary for Tiago, but should be cleaned up in the future)
+                if "grasping_frame" in link_name:
+                    continue
+                # Set poses of meshes relative to the robot to construct the robot
+                link_pose = link_poses[link_name] if link_name in arm_links else self.robot_copy.links_relative_poses[self.robot_copy_type][link_name]
+                mesh_copy_pose = T.pose_transform(*link_pose, *self.robot_copy.relative_poses[self.robot_copy_type][link_name][mesh_name])
+                self._set_prim_pose(copy_mesh, mesh_copy_pose)
+
+    def _set_prim_pose(self, prim, pose):
+        translation = Gf.Vec3d(*np.array(pose[0], dtype=float))
+        prim.GetAttribute("xformOp:translate").Set(translation)
+        orientation = np.array(pose[1], dtype=float)[[3, 0, 1, 2]]
+        prim.GetAttribute("xformOp:orient").Set(Gf.Quatd(*orientation)) 
+
+    def _construct_disabled_collision_pairs(self):
```

Comment:
Does this need to happen here?

Reply:
For most of the disabled collision pairs code, no. However, there is some code that disables collision pairs between the object in the hand of the robot and robot. This code requires the object in the robot hand, which we only know at this point.